gobreaker.go

```diff
@@ -296,8 +298,13 @@ func (cb *CircuitBreaker) onFailure(state State, now time.Time) {
 		if cb.readyToTrip(cb.counts) {
 			cb.setState(StateOpen, now)
 		}
+
 	case StateHalfOpen:
-		cb.setState(StateOpen, now)
+		cb.counts.onFailure()
+		ok, nowOpen := cb.readyToClose(cb.counts)
+		if !ok && nowOpen {
```

Comment:
I think this could be refactored as:
```golang
		if ok, nowOpen := cb.readyToClose(cb.counts); !ok && nowOpen {
			cb.setState(StateOpen, now)
		}
```